lib/multitenancy/headers.js

```diff
@@ -128,7 +128,8 @@ export default function (pluginRoot, server, kbnServer, APP_ROOT, API_ROOT, auth
                 return h.continue;
             }
 
-            const spacesClient = await server.plugins.spaces.getScopedSpacesClient(request);
+            const spaces = server.newPlatform.setup.plugins.spaces;
+            const spacesClient = await spaces.spacesService.scopedClient(request);
```

Comment:
minor: you may want to check if spaces != null before you do `spaces.spacesService.scopedClient(request)`

Reply:
There is a recovery mechanism for this if default space is null and error is logged in the create method